### ISSUE TITLE
Delegate to platform implementations

### ISSUE THREAD

**Issue by penn5** [korlibs/krypto#23]
On the JVM the JCE is available which implements many cryptographical operations. Krypto should avoid using pure-kotlin implementations, but there should be an option to switch back to Kotlin-only in case of a flawed platform.

**Comment by soywiz** [comment]
https://youtrack.jetbrains.com/issue/KT-20427

**Comment by penn5** [comment]
But making an expected extension companion invokation function that returns either the default impl or a custom impl could work.

**Comment by soywiz** [comment]
Yeah, I know. That's what I have been doing all this time. But referenced to do it properly when available.
